staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go

```diff
@@ -982,6 +963,82 @@ func TestList(t *testing.T) {
 	}
 }
 
+func TestListWithoutPaging(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.RemainingItemCount, true)()
+	ctx, store, _ := testSetup(t, withoutPaging())
+
+	_, preset, err := seedMultiLevelData(ctx, store)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	getAttrs := func(obj runtime.Object) (labels.Set, fields.Set, error) {
+		pod := obj.(*example.Pod)
+		return nil, fields.Set{"metadata.name": pod.Name}, nil
+	}
+
+	tests := []struct {
+		name                       string
+		disablePaging              bool
+		rv                         string
+		rvMatch                    metav1.ResourceVersionMatch
+		prefix                     string
+		pred                       storage.SelectionPredicate
+		expectedOut                []*example.Pod
+		expectContinue             bool
+		expectedRemainingItemCount *int64
+		expectError                bool
+	}{
+		{
+			name:          "test List with limit when paging disabled",
```

Comment:
Not strictly related to this PR, but we should increase our test coverage for non-paginating case, I think...